app/graphql/types/input/post/lock.rb

```diff
@@ -0,0 +1,8 @@
+class Types::Input::Post::Lock < Types::Input::Base
+  argument :id, ID, required: true
+  argument :locked_reason, Types::Enum::LockReason, required: true
+
+  def to_model
+    to_h.merge(locked_at: DateTime.current, locked_by: current_user)
+  end
```

Comment:
This feels way too magical for me

Reply:
![kyubey](https://user-images.githubusercontent.com/1283983/97819769-86903280-1c5f-11eb-9c14-0c15590632bc.jpg)




sorry not sorry.